### PR TITLE
[FrameworkBundle] Allow to specify `null` for exception mapping configuration values

### PR DESCRIPTION
**Pull request by andrew-demb**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | 
| License       | MIT
| Doc PR        | 

`null` should be allowed because it is stated as allowed in the configuration info [1][2]
Marked as bugfix due to an existing mismatch between config description and behavior.

[1] https://github.com/symfony/symfony/blob/a2f27add28f7c2c61ef4cfb4e97833509ced0fc0/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L1219-L1220
[2] https://github.com/symfony/symfony/blob/a2f27add28f7c2c61ef4cfb4e97833509ced0fc0/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L1227-L1228

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Fixtures/php/exceptions.php [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
 $container->loadFromExtension('framework', [
     'exceptions' => [
         BadRequestHttpException::class => [
             'log_level' => 'info',
             'status_code' => 422,
         ],
+        NotFoundHttpException::class => [
+            'log_level' => null,
+            'status_code' => 400,
+        ],
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Fixtures/php/exceptions.php:25]: Can we also add a test case when `status_code` set to `null`?

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Fixtures/php/exceptions.php (hunk 1 above)
Comment (andrew-demb) [reply src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Fixtures/php/exceptions.php:25]: Sure, I will add it later

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd [hunk 2]
```diff
@@ -368,7 +368,7 @@
     <xsd:complexType name="exception">
         <xsd:attribute name="name" type="xsd:string" use="required" />
         <xsd:attribute name="log-level" type="xsd:string" />
-        <xsd:attribute name="status-code" type="xsd:int" />
+        <xsd:attribute name="status-code" type="xsd:string" />
```
Comment (andrew-demb) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd]: Probably there may be another way to allow specifying `null` for XML attribute, but I couldn't find any.

Does someone know a more proper way?

File: src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Fixtures/php/exceptions.php (hunk 1 above)
Comment (andrew-demb) [reply src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Fixtures/php/exceptions.php:25]: Done.

Additionally I changed allowed type for XML attribute in XSD schema too to allow specifying `null` https://github.com/symfony/symfony/pull/46956/files#r923095768

**Comment by andrew-demb** [comment]
Rebased on the latest `5.4` to resolve CI failures with doctrine deprecations

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -1227,15 +1227,15 @@ private function addExceptionsSection(ArrayNodeDefinition $rootNode)
...
+                                    ->ifTrue(function ($v) use ($logLevels) { return null !== $v && !\in_array($v, $logLevels, true); })
                                     ->thenInvalid(sprintf('The log level is not valid. Pick one among "%s".', implode('", "', $logLevels)))
                                 ->end()
                                 ->defaultNull()
                             ->end()
                             ->scalarNode('status_code')
                                 ->info('The status code of the response. Null to let Symfony decide.')
                                 ->validate()
-                                    ->ifTrue(function ($v) { return $v < 100 || $v > 599; })
+                                    ->ifTrue(function ($v) { return null !== $v && ($v < 100 || $v > 599); })
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php:1242]: Looking at the code in ErrorListener, we can also use `0` instead of `null`.
Let's fix the info message and allow it instead? That'd fix the XSD.

Comment (andrew-demb) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php:1242]: Done.
Allowed to specify both `null` and `0` to cover XML cases (only `0` is possible to be used) and allow in yaml use preferred format by developers.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd (hunk 2 above)
Comment (andrew-demb) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd]: Not a problem anymore since https://github.com/symfony/symfony/pull/46956#discussion_r982537440

File: src/Symfony/Component/HttpKernel/EventListener/ErrorListener.php [hunk 4]
```diff
@@ -33,8 +33,14 @@ class ErrorListener implements EventSubscriberInterface
     protected $controller;
     protected $logger;
     protected $debug;
+    /**
+     * @var array<class-string, array{log_level: string|null, status_code: int<100,599>|null}>
+     */
     protected $exceptionsMapping;
 
+    /**
+     * @param array<class-string, array{log_level: string|null, status_code: int<100,599>|null}> $exceptionsMapping
```
Comment (andrew-demb) [inline src/Symfony/Component/HttpKernel/EventListener/ErrorListener.php:42]: Is it welcome to explicitly specify typing for mapping structure in Symfony codebase for now?

**Review by nicolas-grekas — APPROVED** [review]
All good on my side, thanks.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @andrew-demb.
